Documentation/concepts/networking.rst

```diff
@@ -189,3 +183,46 @@ this "ingress" load-balancing capability, which achieve the same effect that
 Cilium handles forwarding and security only for ''internal'' traffic between
 different services.
 
+.. _concepts_masquerading:
+
+Masquerading
+============
+
+As the IPv4 addresses used for pods are typically allocated from RFC1918
+private address blocks and are thus not publicly routable. Cilium will
+automatically masquerade the source IP address of all traffic that is leaving
+the cluster to the IPv4 address of the node as the node's IP address is already
+routable on the network.
+
+.. image:: masquerade.png
+    :align: center
+
+This behavior can be disabled with the option ``masquerade: false`` in which
+case no masquerading will be performed.
+
+Configuration
+-------------
+
+Setting the routable CIDR
+  The default behavior is to exclude any destination within the IP allocation
+  CIDR of the local node. If the pod IPs are routable across a wider network,
+  that network can be specified with the option: ``native-routing-cidr:
+  10.0.0.0/8`` in which case all destinations within that CIDR will **not** be
+  masqueraded.
+
+Setting the masquerading interface
+  The default behavior will masquerade all traffic leaving on a non-Cilium
+  network device. This typically leads to the correct behavior. In order to
+  limit the network interface on which masquerading should be performed, the
+  option ``egress-masquerade-interfaces: eth+`` can be used.
+
+Implementation Modes
+--------------------
+
+eBPF-based
+  The eBPF-based implementation is the most efficient
```

Comment:
Once this PR is emerged, I will extend this section to add more info about device selection.